api4/ldap_test.go

```diff
@@ -112,3 +198,43 @@ func TestMigrateIdLdap(t *testing.T) {
 		CheckNotImplementedStatus(t, resp)
 	})
 }
+
+func TestUploadPublicCertificate(t *testing.T) {
+	th := Setup(t)
+	defer th.TearDown()
+
+	// sth.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.SamlSettings.IdpCertificateFile = app.SamlIdpCertificateName })
+	// defer sth.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.SamlSettings.IdpCertificateFile = "" })
+
+	_, resp := th.Client.UploadLdapPublicCertificate([]byte(spPublicCertificate), "pub.crt")
+	require.NotNil(t, resp.Error, "Should have failed. No System Admin privileges")
+
+	_, resp = th.SystemAdminClient.UploadLdapPublicCertificate([]byte(spPublicCertificate), "pub.crt")
+	require.Nil(t, resp.Error, "Should have passed. System Admin privileges %v", resp.Error)
+
+	_, resp = th.Client.DeleteLdapPublicCertificate()
+	require.NotNil(t, resp.Error, "Should have failed. No System Admin privileges")
+
+	_, resp = th.SystemAdminClient.DeleteLdapPublicCertificate()
+	require.Nil(t, resp.Error, "Should have passed. System Admin privileges %v", resp.Error)
+}
+
+func TestUploadPrivateCertificate(t *testing.T) {
+	th := Setup(t)
+	defer th.TearDown()
+
+	// sth.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.SamlSettings.IdpCertificateFile = app.SamlIdpCertificateName })
+	// defer sth.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.SamlSettings.IdpCertificateFile = "" })
+
+	_, resp := th.Client.UploadSamlPrivateCertificate([]byte(spPrivateKey), "private.key")
+	require.NotNil(t, resp.Error, "Should have failed. No System Admin privileges")
+
+	_, resp = th.SystemAdminClient.UploadSamlPrivateCertificate([]byte(spPrivateKey), "private.key")
+	require.Nil(t, resp.Error, "Should have passed. System Admin privileges %v", resp.Error)
```

Comment:
Shall we include local client to the tests? (this question applies to other test casess 😄 )
```suggestion
	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
		_, resp := client.UploadSamlPrivateCertificate([]byte(spPrivateKey), "private.key")
		require.Nil(t, resp.Error, "Should have passed. System Admin privileges %v", resp.Error)
	}
```